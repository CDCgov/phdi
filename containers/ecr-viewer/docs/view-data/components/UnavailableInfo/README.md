[**ecr-viewer**](../../../README.md) â€¢ **Docs**

***

[ecr-viewer](../../../README.md) / view-data/components/UnavailableInfo

# view-data/components/UnavailableInfo

## Index

### Functions

- [default](functions/default.md)
