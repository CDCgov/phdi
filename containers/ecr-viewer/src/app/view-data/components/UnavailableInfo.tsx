import { DisplayData } from "@/app/utils";
import {
  AccordianSection,
  AccordianH3,
  AccordianDiv,
} from "../component-utils";

interface UnavailableInfoProps {
  demographicsUnavailableData: DisplayData[];
  socialUnavailableData: DisplayData[];
  encounterUnavailableData: DisplayData[];
  clinicalUnavailableData: DisplayData[];
  providerUnavailableData: DisplayData[];
  activeProblemsUnavailableData: DisplayData[];
}

const UnavailableInfo = ({
  demographicsUnavailableData,
  socialUnavailableData,
  encounterUnavailableData,
  clinicalUnavailableData,
  providerUnavailableData,
  activeProblemsUnavailableData,
}: UnavailableInfoProps) => {
  const renderRow = (item: any, index: number) => {
    return (
      <div key={index}>
        <div className="grid-row">
          <div className="data-title font-sans-xs">
            <h4>{item.title}</h4>
          </div>
          <div className="grid-col-auto font-sans-xs">N/A</div>
        </div>
        <div className={"section__line_gray"} />
      </div>
    );
  };
  const renderSection = (sectionTitle: string, data: DisplayData[]) => {
    return (
      <div className="margin-bottom-4">
        <h3
          className="usa-summary-box__heading padding-bottom-205"
          id="summary-box-key-information"
        >
          {sectionTitle}
        </h3>
        <div className="usa-summary-box__text">
          {data.map((item, index) => renderRow(item, index))}
        </div>
      </div>
    );
  };

  return (
    <AccordianSection>
      {demographicsUnavailableData?.length > 0 &&
        renderSection("Demographics", demographicsUnavailableData)}
      {socialUnavailableData?.length > 0 &&
        renderSection("Social History", socialUnavailableData)}
      {encounterUnavailableData?.length > 0 &&
        renderSection("Encounter Details", encounterUnavailableData)}
      {clinicalUnavailableData.length > 0 &&
        renderSection("Clinical Info", clinicalUnavailableData)}
      {providerUnavailableData.length > 0 &&
        renderSection("Provider Details", providerUnavailableData)}
      {activeProblemsUnavailableData?.length > 0 && // Add other items under symptoms and problems here
        renderSection("Symptoms and Problems", activeProblemsUnavailableData)}
    </AccordianSection>
  );
};

export default UnavailableInfo;
