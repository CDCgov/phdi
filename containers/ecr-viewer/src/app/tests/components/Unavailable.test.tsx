import { render } from "@testing-library/react";
import { axe } from "jest-axe";
import UnavailableInfo from "../../view-data/components/UnavailableInfo";

describe("UnavailableInfo", () => {
  let container: HTMLElement;
  beforeAll(() => {
    const demographicsUnavailability = [
      {
        title: "Tribal Affiliation",
        value: "",
      },
      {
        title: "Preffered Language",
        value: "",
      },
    ];
    const socialUnavailability = [
      {
        title: "Travel History",
        value: "",
      },
      {
        title: "Pregnancy Status",
        value: "",
      },
      {
        title: "Alcohol Use",
        value: "",
      },
      {
        title: "Sexual Orientation",
        value: "",
      },
      {
        title: "Gender Identity",
        value: "",
      },
    ];
    const encounterUnavailableData = [
      {
        title: "Encounter type",
        value: "",
      },
    ];
    const facilityUnavailableData = [
      {
        title: "Facility Address",
        value: "",
      },
    ];
    const providerUnavailableData = [
      {
        title: "Provider Name",
        value: "",
      },
    ];
    const reasonForVisitUnavailableData = [
      {
        title: "Reason for Visit",
        value: "",
      },
    ];
    const activeProblemsUnavailableData = [
      {
        title: "Problems List",
        value: "",
      },
    ];
    const immunizationsUnavailableData = [
      {
        title: "Immunization History",
        value: "",
      },
    ];
    const vitalUnavailableData = [
      {
        title: "Vitals",
        value: "",
      },
    ];
    const treatmentUnavailableData = [
      {
        title: "Procedures",
        value: "",
      },
    ];
    const clinicalNotesData = [
      {
        title: "Miscellaneous Notes",
        value: "",
      },
    ];
    const ecrMetadata = [
      {
        title: "EHR Software Name",
        value: "",
      },
    ];
    container = render(
      <UnavailableInfo
        demographicsUnavailableData={demographicsUnavailability}
        socialUnavailableData={socialUnavailability}
        encounterUnavailableData={encounterUnavailableData}
        facilityUnavailableData={facilityUnavailableData}
        providerUnavailableData={providerUnavailableData}
        symptomsProblemsUnavailableData={[
          ...reasonForVisitUnavailableData,
          ...activeProblemsUnavailableData,
        ]}
        immunizationsUnavailableData={immunizationsUnavailableData}
        vitalUnavailableData={vitalUnavailableData}
        treatmentData={treatmentUnavailableData}
        clinicalNotesData={clinicalNotesData}
        ecrMetadataUnavailableData={ecrMetadata}
      />,
    ).container;
  });
  it("should match snapshot", () => {
    expect(container).toMatchSnapshot();
  });
  it("should pass accessibility test", async () => {
    expect(await axe(container)).toHaveNoViolations();
  });
});
