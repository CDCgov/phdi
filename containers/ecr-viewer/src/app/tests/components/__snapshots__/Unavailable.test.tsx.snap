// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnavailableInfo should match snapshot 1`] = `
<div>
  <div
    class="margin-top-0"
  >
    <div
      class="padding-bottom-3"
    >
      <div
        class="usa-summary-box__body"
      >
        <div
          class="margin-bottom-4"
        >
          <h3
            class="usa-summary-box__heading padding-bottom-205 unavailable-info"
          >
            Demographics
          </h3>
          <div
            class="usa-summary-box__text"
          >
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Tribal Affiliation
                </div>
                <div
                  class="grid-col maxw7 text-pre-line text-italic text-base"
                >
                  No data
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Preffered Language
                </div>
                <div
                  class="grid-col maxw7 text-pre-line text-italic text-base"
                >
                  No data
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
          </div>
        </div>
        <div
          class="margin-bottom-4"
        >
          <h3
            class="usa-summary-box__heading padding-bottom-205 unavailable-info"
          >
            Social History
          </h3>
          <div
            class="usa-summary-box__text"
          >
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Travel History
                </div>
                <div
                  class="grid-col maxw7 text-pre-line text-italic text-base"
                >
                  No data
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Pregnancy Status
                </div>
                <div
                  class="grid-col maxw7 text-pre-line text-italic text-base"
                >
                  No data
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Alcohol Use
                </div>
                <div
                  class="grid-col maxw7 text-pre-line text-italic text-base"
                >
                  No data
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Sexual Orientation
                </div>
                <div
                  class="grid-col maxw7 text-pre-line text-italic text-base"
                >
                  No data
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Gender Identity
                </div>
                <div
                  class="grid-col maxw7 text-pre-line text-italic text-base"
                >
                  No data
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
          </div>
        </div>
        <div
          class="margin-bottom-4"
        >
          <h3
            class="usa-summary-box__heading padding-bottom-205 unavailable-info"
          >
            Encounter Details
          </h3>
          <div
            class="usa-summary-box__text"
          >
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Encounter type
                </div>
                <div
                  class="grid-col maxw7 text-pre-line text-italic text-base"
                >
                  No data
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
          </div>
        </div>
        <div
          class="margin-bottom-4"
        >
          <h3
            class="usa-summary-box__heading padding-bottom-205 unavailable-info"
          >
            Facility Details
          </h3>
          <div
            class="usa-summary-box__text"
          >
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Facility Address
                </div>
                <div
                  class="grid-col maxw7 text-pre-line text-italic text-base"
                >
                  No data
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
          </div>
        </div>
        <div
          class="margin-bottom-4"
        >
          <h3
            class="usa-summary-box__heading padding-bottom-205 unavailable-info"
          >
            Clinical Notes
          </h3>
          <div
            class="usa-summary-box__text"
          >
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Miscellaneous Notes
                </div>
                <div
                  class="grid-col maxw7 text-pre-line text-italic text-base"
                >
                  No data
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
          </div>
        </div>
        <div
          class="margin-bottom-4"
        >
          <h3
            class="usa-summary-box__heading padding-bottom-205 unavailable-info"
          >
            Provider Details
          </h3>
          <div
            class="usa-summary-box__text"
          >
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Provider Name
                </div>
                <div
                  class="grid-col maxw7 text-pre-line text-italic text-base"
                >
                  No data
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
          </div>
        </div>
        <div
          class="margin-bottom-4"
        >
          <h3
            class="usa-summary-box__heading padding-bottom-205 unavailable-info"
          >
            Symptoms and Problems
          </h3>
          <div
            class="usa-summary-box__text"
          >
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Reason for Visit
                </div>
                <div
                  class="grid-col maxw7 text-pre-line text-italic text-base"
                >
                  No data
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Problems List
                </div>
                <div
                  class="grid-col maxw7 text-pre-line text-italic text-base"
                >
                  No data
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
          </div>
        </div>
        <div
          class="margin-bottom-4"
        >
          <h3
            class="usa-summary-box__heading padding-bottom-205 unavailable-info"
          >
            Diagnostics and Vital Signs
          </h3>
          <div
            class="usa-summary-box__text"
          >
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Vitals
                </div>
                <div
                  class="grid-col maxw7 text-pre-line text-italic text-base"
                >
                  No data
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
          </div>
        </div>
        <div
          class="margin-bottom-4"
        >
          <h3
            class="usa-summary-box__heading padding-bottom-205 unavailable-info"
          >
            Immunizations
          </h3>
          <div
            class="usa-summary-box__text"
          >
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Immunization History
                </div>
                <div
                  class="grid-col maxw7 text-pre-line text-italic text-base"
                >
                  No data
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
          </div>
        </div>
        <div
          class="margin-bottom-4"
        >
          <h3
            class="usa-summary-box__heading padding-bottom-205 unavailable-info"
          >
            Treatment Details
          </h3>
          <div
            class="usa-summary-box__text"
          >
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Procedures
                </div>
                <div
                  class="grid-col maxw7 text-pre-line text-italic text-base"
                >
                  No data
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
          </div>
        </div>
        <div
          class="margin-bottom-4"
        >
          <h3
            class="usa-summary-box__heading padding-bottom-205 unavailable-info"
          >
            eCR Metadata
          </h3>
          <div
            class="usa-summary-box__text"
          >
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  EHR Software Name
                </div>
                <div
                  class="grid-col maxw7 text-pre-line text-italic text-base"
                >
                  No data
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Custodian Contact
                </div>
                <div
                  class="grid-col maxw7 text-pre-line text-italic text-base"
                >
                  No data
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
